File: gpfksdlrn/level1/최소직사각형.js
// https://school.programmers.co.kr/learn/courses/30/lessons/86491

function solution(sizes) {
  const [width, height] = sizes.reduce(
    ([maxW, maxH], [w, h]) => [
      Math.max(maxW, Math.max(w, h)),
      Math.max(maxH, Math.min(w, h)),
    ],
    [0, 0],
  );

  return width * height;
}

File: gnsals0904/level1/최소_직사각형.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int width = 0;
        int length = 0;
        for(int[] size : sizes){
            width = Math.max(Math.max(size[0], size[1]), width);
            length = Math.max(Math.min(size[0], size[1]), length);
        }
        int answer = width * length;
        return answer;
    }
}